[aspnetcore] Add built-in signalR functionality

KrikTze opened the pull request:
[platform] - Add built-in signalR functionality to projects.

travlos committed:
Add SignalR endpoint support with Azure SignalR SDK

Introduced SignalR endpoint support using the Azure SignalR SDK:
- Added `Microsoft.Azure.SignalR.Management` package.
- Created `SignalrEndpointApi` for adding and mapping SignalR endpoints.
- Implemented `AddSignalREndpoints` and `MapSignalrEndpoints` methods.
- Added `SignalREndpointsOptions` for endpoint configuration.
- Introduced `SignalREndpointsHandler` for handling SignalR operations:
  - `Negotiate` for generating negotiation responses.
  - `BroadcastToUser` and `BroadcastToUsers` for message broadcasting.
- Added supporting record types: `NegotiateResponse` and `BroadcastCommand`.
- Updated endpoint mapping with OpenAPI security and authorization.

KrikTze committed:
Add serverless signalr functionality to any project
- Add 2 interfaces /services- for broadcasting and for listening
- Enhance parameterization fo SignalROptions
- Use concurrent dictionary + Singleton service to use existing ContextHub

travlos committed:
Merge branch 'develop' into feature/AspNetCore/SignalrEndpoint

KrikTze committed:
Apply some refactoring to SignalR servicis - and Add constraints like allowedHubs

KrikTze committed:
Rename Files - Add comments - Fix format - Remove unnecessary variables

KrikTze committed:
Remove unused import

travlos commented on src/Indice.AspNetCore/Features/SignalREndpoint/SignalRBroadcastService.cs:
```diff
@@ -0,0 +1,55 @@
+            .User(userId)
+                .SendCoreAsync(
+                    method: "broadcastMessage",
+                    args: ["system", command.Message],
```
remove system and get method from command

cleftheris committed:
Merge branch 'develop' into feature/AspNetCore/SignalrEndpoint

KrikTze committed:
Make method part of the Broadcast Command,
Add cancellation token to hubContext

cleftheris committed:
Merge branch 'develop' into feature/AspNetCore/SignalrEndpoint

KrikTze committed:
Merge branch 'develop' into feature/AspNetCore/SignalrEndpoint

KrikTze committed:
Merge branch 'develop' into feature/AspNetCore/SignalrEndpoint

KrikTze committed:
Rename signalR
Temporarily add back broadcast

KrikTze committed:
Add addUser to groups - end point
The service should stay , but the endpoint should probably be removed

cleftheris committed:
Merge branch 'develop' into feature/AspNetCore/SignalrEndpoint

cleftheris committed:
Refactor SignalR endpoint to modular SignalRProxy feature

Moved and renamed SignalR endpoint code to new SignalRProxy feature under Indice.AspNetCore.Features.SignalRProxy and Indice.Services. Updated all related classes, interfaces, and service registrations for improved modularity and extensibility. Enhanced configuration options, OpenAPI documentation, and authorization policies. Service registration now uses new extension methods. SignalR Azure SDK dependency moved to Indice.Services. Old SignalREndpoint code removed and replaced with new structure.

cleftheris committed:
Refactor SignalR negotiation service and add auth helpers

Renamed and refactored the SignalR negotiation service interface and implementation for naming consistency (ISignalRProxyNegotiatiationService). Updated all usages and DI registrations accordingly. Introduced SignalRProxyAuthentication static class with helpers for negotiation authentication, including a custom header and policy scheme. Cleaned up using statements and improved code structure.

KrikTze committed:
Add descriptions and summaries

cleftheris committed:
Add SignalR Proxy feature and improve authentication docs

Introduces SignalR Proxy for offloading connections and management endpoints. Updates README with usage instructions and workaround for old SDK authentication issues. Registers negotiation authentication scheme for proper token handling.

cleftheris committed:
Merge branch 'feature/AspNetCore/SignalrEndpoint' of https://github.com/indice-co/Indice.Platform into feature/AspNetCore/SignalrEndpoint

cleftheris committed:
Refactor SignalR proxy auth; add legacy support method

Refactored authentication setup to use local configuration methods for JWT Bearer and OAuth2 Introspection. Introduced AddSignalRProxyLagacyAuthentication for legacy SignalR scenarios, supporting both JWT and reference tokens. Removed the old AddSignalRProxyAuthentication policy scheme method. Updated README and added necessary usings. Centralizes and modernizes SignalR proxy authentication configuration.

KrikTze committed:
Add legacy scheme for signalR auth

cleftheris committed:
Fix typo in AddSignalRProxyLegacyAuthentication method name

Corrected "Lagacy" to "Legacy" in the method name within WebApplicationBuilderExtensions to ensure proper usage and consistency. No functional changes were made.

cleftheris committed:
Bump base version to 8.32; reset component versions to .0

Updated VersionPrefixBase from 8.31 to 8.32. Reset Identity, IdentityUI, Cases, Messages, and Media version prefixes to .0 to align with the new base version.

cleftheris reviewed (approved):
🚀

travlos reviewed (approved):
👍

cleftheris commented:
@spapoutsis @erotokritosVall  this feature is the one you are interested in. It is rolling out today in `v8.32` 

cc @dkarkanas 🚀
